### PR TITLE
increase l1 cluster threshold from 2000 to 4000e

### PR DESCRIPTION
**Pull request by dkotlins**
#### PR description:

Increase L1 cluster threshold from 2000 to 4000e making it the same as all other layers.
With the new L1 detector the lower thresholds should not be needed anymore.
The change will affect the cluster charge distribution for L1, the lower edge will shift from 2000
to 4000e.

#### PR validation:

Run RECO on raw data from the cosmic run 344420.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport needed.

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by mmusich** [comment]
what about:

https://github.com/cms-sw/cmssw/blob/a686932f1b9c01341d8936a04bb95ca540308772/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h#L12

?

**Comment by dkotlins** [comment]
Who has added this? I though we were suppose to avoid hardwired constants. How does the CMSSW code checking 
accept something like this?

I can change it as well, but I a, not even sure how it works.

Danek

**Comment by VinInn** [comment]
it is configurable
https://cmssdt.cern.ch/dxr/CMSSW/search?q=kSiPixelClusterThresholdsDefaultPhase1&case=true&redirect=true

File: RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h [hunk 1]
```diff
@@ -9,6 +9,6 @@ struct SiPixelClusterThresholds {
   const int32_t otherLayers;
 };
 
-constexpr SiPixelClusterThresholds kSiPixelClusterThresholdsDefaultPhase1{.layer1 = 2000, .otherLayers = 4000};
+constexpr SiPixelClusterThresholds kSiPixelClusterThresholdsDefaultPhase1{.layer1 = 4000, .otherLayers = 4000};
```
Comment (VinInn) [inline RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h:12]: we need a modifier for Run2 then

Comment (mmusich) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h:12]: It needs to be configured @dkotlins not change the default

Comment (VinInn) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h:12]: what about older Run3 Relvals?

**Comment by mmusich** [comment]
> it is configurable
https://cmssdt.cern.ch/dxr/CMSSW/search?q=kSiPixelClusterThresholdsDefaultPhase1&case=true&redirect=true

Yep, that was my point.  Currently takes it form `fillDescriptions` the default

**Comment by dkotlins** [comment]
Does this mean that we cannot change anymore any of the reconstruction parameters?

**Comment by mmusich** [comment]
> this mean that we cannot change anymore any of the reconstruction parameters?

No, of course you can

**Comment by czangela** [comment]
One should be able to set from the config, as in the [cfipython file].(https://cmssdt.cern.ch/dxr/CMSSW/source/cfipython/RecoLocalTracker/SiPixelClusterizer/siPixelRawToClusterCUDA_cfi.py#8).

File: RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h (hunk 1 above)
Comment (mmusich) [reply RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterThresholds.h:12]: Too bad

**Comment by ferencek** [comment]
@dkotlins, what's then left to be done is what @mmusich proposed in https://mattermost.web.cern.ch/cms-exp/pl/6w9qpr3p1fdgpd8o6j9ih5bfmc Effectively it means undoing https://github.com/cms-sw/cmssw/pull/35506/commits/ecdff58820cd7783c84ddfeadbda86dfcea5d342 and adding/updating run3 modifiers in https://github.com/cms-sw/cmssw/blob/CMSSW_12_1_0_pre3/RecoLocalTracker/SiPixelClusterizer/python/siPixelClustersPreSplitting_cff.py

**Comment by dkotlins** [comment]
seems very complicated. Will get back to it later.

**Pull request closed without merging**